### PR TITLE
Add some modules that fulfil Computer Science requirements

### LINKED ISSUE #120: Hardcode modules linked to majors
List the modules needed to fulfil a major, and what requirements each module fulfils

### PR DESCRIPTION
**Pull request by GabrielYik**
Resolves #120 

**Legend**
With reference to https://www.comp.nus.edu.sg/cugresource/per-cohort/cs/cs-17-18/
PR_F : Programme Requirements - Foundation
PR_IER : Programme Requirements - Industrial Experience Requirement
PR_MS_M : Programme Requirements - Mathematics & Sciences - Mathematics
PR_BD_FA_P_SE : Programme Requirements - Breadth & Depth - Focus Area - Primaries - Software Engineering
...

**Major Supported**
* Computer Science

**Focus Areas Supported**
* Software Engineering
* Algorithms & Theory
* Artificial Intelligence

**Note**
All modules implicitly are unrestricted electives (except the core modules)

### CODE REVIEW

File: src/main/resources/data/computerscience.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+      { "name": "CS3230", "types": ["PR_IER"] },
+      { "name": "CP3880", "types": ["PR_IER"] },
+      { "name": "CP3200", "types": ["PR_IER"] },
+      { "name": "CP3202", "types": ["PR_IER"] },
+      { "name": "IS4010", "types": ["PR_IER"] },
+      { "name": "CP3200", "types": ["PR_IER"] },
+      { "name": "CP3107", "types": ["PR_IER"] },
+
+      { "name": "IS1103/X" , "types": ["PR_IT"] },
```
Comment (rongjiecomputer) [inline src/main/resources/data/computerscience.json]: There are no modules named `IS1103/X` or `IS1103X`, just `IS1103`.

File: src/main/resources/data/computerscience.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      { "name": "CS2101", "types": ["PR_IT"] },
+      { "name": "ES2660", "types": ["PR_IT"] },
+
+      { "name": "MA1521", "types": ["PR_MS_M"] },
+      { "name": "MA1101R", "types": ["PR_MS_M"] },
+      { "name": "ST2334", "types": ["PR_MS_M"] },
+      { "name": "PC1221/X", "types": ["PR_MS_S"] },
```
Comment (rongjiecomputer) [inline src/main/resources/data/computerscience.json]: Break this into 2:

```suggestion
      { "name": "PC1221", "types": ["PR_MS_S"] },
      { "name": "PC1221X", "types": ["PR_MS_S"] },
```

File: src/main/resources/data/computerscience.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+      { "name": "CS3203", "types": ["PR_BD"] },
+      { "name": ["CS3216", "CS3217"], "types": ["PR_BD"] },
```
Comment (rongjiecomputer) [inline src/main/resources/data/computerscience.json]: Why `name` is an array?

File: src/main/resources/data/computerscience.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      { "name": ["CS3281", "CS3282"], "types": ["PR_BD"] },
+
+      { "name": "GEH", "types": ["ULR"] },
+      { "name": "GER", "types": ["ULR"] },
+      { "name": "GET", "types": ["ULR"] },
+      { "name": "GEQ", "types": ["ULR"] },
+      { "name": "GES", "types": ["ULR"] },
```
Comment (rongjiecomputer) [inline src/main/resources/data/computerscience.json]: Every major takes `GE` modules as ULR. These 5 codes are not valid module codes by themselves. Let's keep `name` to always be valid module code that you can match with `ModuleInfo.getFromModuleCode`. We can code GE module's logic separately.

File: src/main/resources/data/computerscience.json (hunk 3 above)
Comment (GabrielYik) [reply src/main/resources/data/computerscience.json]: Those two modules are meant to be taken together, so I put them together

File: src/main/resources/data/computerscience.json (hunk 4 above)
Comment (GabrielYik) [reply src/main/resources/data/computerscience.json]: Oops sorry forgot to add in the numbers

**Pull request merged**